samza-core/src/main/scala/org/apache/samza/metrics/JvmMetrics.scala

```diff
@@ -63,6 +68,10 @@ class JvmMetrics(val registry: MetricsRegistry) extends MetricsHelper with Runna
   val cGcCount = newCounter("gc-count")
   val cGcTimeMillis = newCounter("gc-time-millis")
 
+  val gProcessCpuUsage = if (osMXBean.isInstanceOf[OperatingSystemMXBean]) newGauge("process-cpu-usage", 0.0) else null
```

Comment:
Do we need these metrics to be null? Or, Do you think it's convenient to report zero values for metrics instead of not reporting them at all?

Reply:
I think it's less confusing to not emit the metrics if they're not functional.

Makes for ugly code, but a better user experience.